frontend/src/components/EditElectionRoll.tsx

```diff
@@ -6,15 +6,47 @@ import Typography from '@material-ui/core/Typography';
 import Divider from '@material-ui/core/Divider';
 import Container from '@material-ui/core/Container';
 import { Paper, Table, TableBody, TableCell, TableContainer, TableHead, TableRow, Tooltip } from "@material-ui/core";
+import useFetch from "../useFetch";
+import { ElectionRollState } from "../../../domain_model/ElectionRoll";
 
-const EditElectionRoll = ({ roll, onClose }) => {
+const EditElectionRoll = ({ roll, onClose, fetchRolls, id}) => {
     const [updatedRoll, setUpdatedRoll] = useState(roll)
+
+    const approve = useFetch(`/API/Election/${id}/rolls/approve`,'post')
+    const flag = useFetch(`/API/Election/${id}/rolls/flag`,'post')
+    const unflag = useFetch(`/API/Election/${id}/rolls/unflag`,'post')
+    const invalidate = useFetch(`/API/Election/${id}/rolls/invalidate`,'post')
+
+    const onApprove = async () => {
```

Comment:
theoretically can combine these into a single "request", with a type (validate, flag, unflag, etc) that determines the URL.  Then elsewhere in the page don't need to check for 4 different responses or errors, just of the active "request"